### PR TITLE
feat: convert data_getting_started docs

### PR DESCRIPTION
**Pull request by MarkComerford**
#### 📲 What

Converted the data getting started docs to asciidoc

#### 🤔 Why
		
Why it's needed, background context.
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 Evidence
		
Screenshots / external resources / links / etc.
Link to documentation updated with changes impacted in the PR.
		 
#### 🕵️ How to test

Notes on how a reviewer can test the changes, e.g. how to run the tests.

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    * `AZURE_STORAGE_ACCOUNT_NAME`
+    * `AZURE_CONFIG_ACCOUNT_NAME`
+
+== Running tests
+
+`make` commands are provided to assist with running tests while developing locally. See link:../etl_pipelines/testing_data_azure.adoc[testing] for further details on these tests.
+
+[TIP]
+If you encounter PATH-related issues with Poetry when running the tests, we recommend installing Poetry using
+[pipx](https://python-poetry.org/docs/#installing-with-pipx) rather than the official installer.
```
Comment (isaac-shek) [inline asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc:38]: link text

File: asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+Running the end-to-end tests will involve executing Data Factory pipelines in Azure. Ensure you have set up the link:#optional-azure-connection[Azure connection] and run:
+
+[source,bash]
+----
+make test_e2e
+----
+
+[TIP]
+Running end-to-end tests from your local machine may require additional permissions in Azure. If the tests fail whilst clearing up directories, ensure that you have [Storage Blob Data Contributor](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#storage-blob-data-contributor) access applied to your Azure Active Directory subscription. You may also be required to configure the [firewall rules](https://learn.microsoft.com/en-us/azure/storage/common/storage-network-security) for the storage account to whitelist your IP address.
```
Comment (isaac-shek) [inline asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc:59]: link text

File: asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc (hunk 1 above)
Comment (MarkComerford) [reply asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc]: fixed

File: asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc (hunk 2 above)
Comment (MarkComerford) [reply asciidoc/workloads/azure/data/getting_started/dev_quickstart_data_azure.adoc]: fixed

**Review by isaac-shek — APPROVED** [review]

**Pull request merged**